### ISSUE TITLE
Headers ignore the `extend` font face attribute

### ISSUE THREAD

**Issue by cyrus-and** [jrblevin/markdown-mode#705]
## Expected Behavior

The `extend` attribute should be respected, just like it happens for `markdown-code-face`. That is, the background color of the `markdown-header-face-*` should extend to the whole width of the buffer when `extend` is `t`.

## Actual Behavior

The background does not extend:

<img width="803" alt="Screenshot 2022-05-06 at 10 41 24 AM" src="https://user-images.githubusercontent.com/1660263/167098002-6183f35a-db6a-49c5-98e0-b2285c406d05.png">

## Steps to Reproduce

1. Place the following in `/tmp/emacs/.emacs`:

    ```el
    (setq custom-file "/dev/null")

    (package-initialize)
    (add-to-list 'package-archives '("melpa" . "https://melpa.org/packages/"))

    (package-install 'markdown-mode)

    (custom-set-faces
     `(markdown-header-face-1 ((t (:background "#ff0000" :extend t)))))
    ```

2. Start Emacs with `HOME=/tmp/emacs emacs`;

3. Create or open a Markdown file with at least one top-level header:

    ~~~md
    # Header

    ```
    code
    ```
    ~~~

## Backtrace

There are no errors.

## Software Versions

- Markdown Mode: version 2.6-dev
- Emacs: GNU Emacs 29.0.50 (build 1, x86_64-apple-darwin20.6.0, NS appkit-2022.60 Version 11.6.5 (Build 20G527)) of 2022-04-29
- OS: macOS Big Sur 11.6.5

**Comment by cyrus-and** [comment]
> BTW `markdown-asymmetric-header` is used only for inserting headers, so I suppose it cannot be used for fontification flag.

Yeah, it makes sense, I was only suggesting that in case you didn't want to introduce a new variable.

**Comment by syohex** [comment]
@cyrus-and How about #706 ?

**Comment by cyrus-and** [comment]
> @cyrus-and How about #706 ?

Just tried it and seems to work perfectly! Thank you.

**Comment by syohex** [comment]
I have merged #706
